src/components/accounts/RecoverAccountSeedPhrase.js

```diff
@@ -2,96 +2,98 @@ import React, { Component } from 'react'
 import { withRouter } from 'react-router-dom'
 import { connect } from 'react-redux'
 
-import { recoverAccountSeedPhrase, redirectToApp, checkAccountAvailable, clear } from '../../actions/account'
+import { recoverAccountSeedPhrase, redirectToApp, checkAccountAvailable, clear, handleRefreshAccount } from '../../actions/account'
 
 import RecoverAccountSeedPhraseForm from './RecoverAccountSeedPhraseForm'
 import AccountFormSection from './AccountFormSection'
 import AccountFormContainer from './AccountFormContainer'
 
 class RecoverAccountSeedPhrase extends Component {
-   state = {
-      accountId: '',
-      seedPhrase: '',
-      isLegit: false
-   }
+    state = {
+        accountId: '',
+        seedPhrase: '',
+        isLegit: false
+    }
 
-   componentDidMount = () => {}
+    componentDidMount = () => {}
 
-   componentWillUnmount = () => {
-      this.props.clear()
-   }
+    componentWillUnmount = () => {
+        this.props.clear()
+    }
 
-   handleChange = (e, { name, value }) => {
-      this.setState((state) => ({
-         [name]: value,
-         isLegit: name === 'accountId' ? state.isLegit : this.isLegitField(name, value)
-      }))
-   }
+    handleChange = (e, { name, value }) => {
+        this.setState((state) => ({
+            [name]: value,
+            isLegit: name === 'accountId' ? state.isLegit : this.isLegitField(name, value)
+        }))
+    }
 
-   isLegitField(name, value) {
-      // TODO: Use some validation framework?
-      let validators = {
-         seedPhrase: value => true // TODO validate seed phrase
-      }
-      return validators[name](value);
-   }
+    isLegitField(name, value) {
+        // TODO: Use some validation framework?
+        let validators = {
+            seedPhrase: value => true // TODO validate seed phrase
+        }
+        return validators[name](value);
+    }
 
-   handleSubmit = e => {
-      e.preventDefault()
+    handleSubmit = e => {
+        e.preventDefault()
 
-      if (!this.state.isLegit) {
-         return false
-      }
+        if (!this.state.isLegit) {
+            return false
+        }
 
-      const accountId = this.state.accountId
-      this.props.recoverAccountSeedPhrase(this.state.seedPhrase, accountId)
-         .then(({ error }) => {
-            if (error) return
-            this.props.redirectToApp()
-         })
-         .finally(() => {
-            this.setState(() => ({
-               isLegit: false
-            }))
-         })
-   }
+        const accountId = this.state.accountId
+        this.props.recoverAccountSeedPhrase(this.state.seedPhrase, accountId)
+            .then(({ error }) => {
+                if (error) return
+                this.props.handleRefreshAccount()
+                this.props.redirectToApp()
```

Comment:
not sure I fully understand the logic of the fix. What will happen on `redirectToApp()`, will it always just short-circuit to `/authorized-apps`, etc?

Reply:
The problem with #323 was that during redirection (after account is recovered) the Redux state is not updated yet with accountId, so it's redirecting again to `/create`. Adding `handleRefreshAccount()` before redirection will fix the problem.

Changes might look weird because of changing indention but I only add `handleRefreshAccount` before `redirectToApp()`.

Reply:
What will happen in such scenario:
1) App requests login using NEAR Wallet (i.e. redirects to wallet with all stuff specified in URL)
2) There is no account yet, so user is redirected to /create page where there is option to recover.
3) User choses to recover.

Basically, will original redirect intent get preserved?

Reply:
Yes, after recovery, user will be redirected to `/login` and will be able to proceed.

Reply:
Awesome, I’ll test and merge later!